### PR TITLE
added some useful validators

### PR DESCRIPTION
**Pull request by 0xch4z**
Just thought I'd add some things that I used in a side project to get to know vapor. Let me know if I need to change anything

### CODE REVIEW

File: Tests/ValidationTests/ValidationTests.swift [hunk 1]
```diff
@@ -78,4 +78,59 @@ class ValidationTests: XCTestCase {
...
+        XCTAssertTrue("VmFwb3IgPiBFeHByZXNz".passes(Base64Validator()))
+        XCTAssertFalse("bWFsZmalformedl9ybWVk".passes(Base64Validator()))
+        XCTAssertFalse("".passes(Base64Validator()))
+    }
+
+    func testValidASCII() {
+        XCTAssertTrue(" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~".passes(ASCIIValidator()))
+        XCTAssertFalse("piędziesięciogroszówka".passes(ASCIIValidator()))
+    }
+
```
Comment (tanner0101) [inline Tests/ValidationTests/ValidationTests.swift:145]: there should be a `static let allTests` somewhere in this doc. can you add these tests to it? (without that, they won't run on linux). thanks!

**Review by tanner0101 — CHANGES_REQUESTED** [review]

**Comment by 0xch4z** [comment]
Yikes... added the email validation (not my commit) tests to the test array and its failing, gonna try to figure out why and revert. It passed on OS X, Idk what happened on linux.

EDIT:
```swift
print(EmailValidator().validate("æøå@gmail.com") == false)
```
on Linux, this returns false, on OS X it returns true. Not really sure why that is, I'm just going to clean the validator up with some regex instead of the current logic.

**Comment by 0xch4z** [comment]
It passed 😅. Really enjoying the Vapor project, and I'd like to help out in anyway possible. I know you guys are busy with documentation and such for 2.0, so if its wanted, I'm happy to add a lot more validators and refactor some of the current ones. Let me know

**Review by tanner0101 — APPROVED** [review]

**Comment by tanner0101** [comment]
Thanks @Charliekenney23! Any documentation you'd like to submit regarding validation would be greatly appreciated https://github.com/vapor/documentation

**Pull request merged**